.github/workflows/ci-build-test.yml

```diff
@@ -18,12 +34,12 @@ jobs:
 
     steps:
     - name: Clone the repo
-      uses: actions/checkout@v4
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
I don't think I've seen pinning to a particular SHA before. Seems like something we would do to account for a particular issue/regression we've been encountering in a particular action.